langchain4j-redis/src/main/java/dev/langchain4j/store/embedding/redis/RedisEmbeddingStoreImpl.java

```diff
@@ -0,0 +1,174 @@
+package dev.langchain4j.store.embedding.redis;
+
+import com.google.gson.Gson;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.internal.ValidationUtils;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import redis.clients.jedis.JedisPooled;
+import redis.clients.jedis.exceptions.JedisDataException;
+import redis.clients.jedis.search.*;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static dev.langchain4j.internal.Utils.isCollectionEmpty;
+import static dev.langchain4j.internal.Utils.randomUUID;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static redis.clients.jedis.search.RediSearchUtil.ToByteArray;
+
+/**
+ * Redis Embedding Store Implementation
+ */
+public class RedisEmbeddingStoreImpl implements EmbeddingStore<TextSegment> {
+
+    private static final Logger log = LoggerFactory.getLogger(RedisEmbeddingStoreImpl.class);
+    private static final Gson GSON = new Gson();
+
+    private final JedisPooled client;
+    private final RedisSchema schema;
+
+    public RedisEmbeddingStoreImpl(String url, Integer dimension) {
+        url = ensureNotNull(url, "url");
+        dimension = ensureNotNull(dimension, "dimension");
+
+        client = new JedisPooled(url);
+        schema = RedisSchema.builder().dimension(dimension).build();
+
+        createIndex(schema.getIndexName());
```

Comment:
can you move `isIndexExist()` check out of `createIndex()` method?
```java
if (!isIndexExist(..)) {
  createIndex(..);
}
```

Reply:
done